File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = list()
        right = [0] * len(nums)

        product = 1
        for x in nums:
            left.append(product)
            product *= x

        product = 1
        for i in range(len(nums)-1,-1,-1):
            right[i] = left[i]*product
            product *= nums[i]

        return right 

        
        

File: Data Structures & Algorithms/rotate-array/submission-1.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length-1;
        k = k%(n+1);
        reverse(nums,0,n);
        reverse(nums,0,k-1);
        reverse(nums,k,n);
    }

    private void reverse(int arr[], int left, int right) {
        while(right > left) {
            int temp = arr[left];
            arr[left] = arr[right];
            arr[right] = temp;
            right--;
            left++;
        }
    }
}